app/views/posts/index.html.erb

```diff
@@ -1 +1,33 @@
-<h1> This is posts page index </h1>
\ No newline at end of file
+<h1> This is posts page index </h1>
+
+<div class="wrapper-1">
+    <li class="item">  
+       <img class="photo" src="<%= @user.photo%>" alt="user photo">
+       <div class= "container"> 
+          <div class="name"> <%= @user.name %> </div>
+          <div class="post-num"> number of posts: <%= @user.posts_counter %></div>
+       </div>
+    </li>
+  
+    <% @posts.each do |post| %>
+      <div class= "container-column-2"> 
+          <div class="item">Post # <%= post.id %>  <%= post.title %> </div>
+          <div class="item"> <%= post.text %> </div>
+
+          <div class="container-2">               
+            <div>Comments: <%= post.comments_counter %> </div>
+            <div>likes: <%= post.likes_counter %> </div>
+          </div>
+      </div>
+  
+      <div class= "container-column-2"> 
+        <% com =  @comments.where(post_id: post.id) %>
+        <% if com!= nil  %>
+            <% com.each do |comment|  %>
+            <div> <%= @users.find(comment.author_id).name %>:  <%= comment.text %> </div>
+            <% end %>
+        <% end %>        
+      </div>
+    <% end %>
```

Comment:
- [ ] It is required that a post should render 5 most recent comments. This is not so in your case. Kindly look into it and fix.

You can check the code below for some insights.

```
   <% post.recent_comments.each do |comment| %>
       <p><%= comment.author.name %>: <%= comment.text %></p> 
    <% end %>
```